src/skins/vector/views/molecules/SearchBar.js

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2015 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+var React = require('react');
+var MatrixClientPeg = require('matrix-react-sdk/lib/MatrixClientPeg');
+var sdk = require('matrix-react-sdk');
+
+module.exports = React.createClass({
+    displayName: 'SearchBar',
+
+    Scope : {
+        Room: 'Room',
+        All: 'All',
+    },
+
+    getInitialState: function() {
+        return ({
+            scope: this.Scope.Room
+        });
+    },
+
+    onThisRoomClick: function() {
+        this.setState({ scope: this.Scope.Room });
+    },
+
+    onAllRoomsClick: function() {
+        this.setState({ scope: this.Scope.All });
+    },
+
+    onSearchChange: function(e) {
+        if (e.keyCode === 13) {
```

Comment:
Please either comment what key code `13` is, or put a constant to explain it instead.